pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -532,12 +532,19 @@ func (e *Controller) syncService(ctx context.Context, key string) error {
 	}
 
 	logger.V(4).Info("Update endpoints", "service", klog.KObj(service), "readyEndpoints", totalReadyEps, "notreadyEndpoints", totalNotReadyEps)
+	shouldFlagRVAsStale := false
 	if createEndpoints {
 		// No previous endpoints, create them
 		_, err = e.client.CoreV1().Endpoints(service.Namespace).Create(ctx, newEndpoints, metav1.CreateOptions{})
 	} else {
 		// Pre-existing
-		_, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		var updatedEndpoints *v1.Endpoints
+		updatedEndpoints, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		// an webhook could potentially cause endpoints update operation became noop and return same resourceVersion.
+		// see https://github.com/kubernetes/kubernetes/issues/126578#issuecomment-2314368183
+		if updatedEndpoints != nil && updatedEndpoints.ResourceVersion != currentEndpoints.ResourceVersion {
+			shouldFlagRVAsStale = true
```

Comment:
how about naming it `shouldStaleEndpoints` to align with the `Stale(currentEndpoints)` operation?

Reply:
`shouldTrackStaleEndpoints`  reads better IMHO

Reply:
sounds good, i can update it to be `shouldTrackStaleEndpoints`